### PR TITLE
cpdp-1241-1228-1229-1266-enter-one-property-details: Multiple Disposal Enter one property details

### PR DESCRIPTION
**Pull request by rambabu-posa**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala [hunk 1]
```diff
@@ -183,6 +199,234 @@ class PropertyDetailsController @Inject() (
...
+                          answers
+                            .getOrElse(IncompleteMultipleDisposalsExamplePropertyDetailsAnswers.empty)
+                            .fold(
+                              _.copy(disposalDate = Some(disposalDate)),
+                              _.copy(disposalDate = disposalDate)
+                            )
+                        val updatedDraftReturn = m.copy(examplePropertyDetailsAnswers = Some(updatedAnswers))
+                        val result = for {
+                          _ <- EitherT(
+                                updateSession(sessionStore, request)(
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala]: you need to call the returnService to update the draft return in the BE as well

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala [hunk 2]
```diff
@@ -183,6 +199,234 @@ class PropertyDetailsController @Inject() (
...
+
+            disposalPriceForm
+              .bindFromRequest()
+              .fold(
+                formWithErrors =>
+                  BadRequest(
+                    multipleDisposalsDisposalPricePage(formWithErrors, backLink)
+                  ), { disposalPrice =>
+                  if (answers
+                        .getOrElse(IncompleteMultipleDisposalsExamplePropertyDetailsAnswers.empty)
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala]: again, you can stick `.getOrElse` on the `answers` val above

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala [hunk 3]
```diff
@@ -238,6 +491,40 @@ class PropertyDetailsController @Inject() (
...
+    val maximumDateInclusive = if (endDateOfTaxYear.isAfter(today)) endDateOfTaxYear else today
+
+    disposalDateForm(maximumDateInclusive, startDateOfTaxYear)
+  }
+
+  private def disposalDateBackLink(answers: Option[MultipleDisposalsExamplePropertyDetailsAnswers]): Call =
+    answers
+      .getOrElse(IncompleteMultipleDisposalsExamplePropertyDetailsAnswers.empty)
+      .fold(
+        _ => routes.PropertyDetailsController.enterUkAddress(),
```
[the remark this answers is not in the document]
Comment (rambabu-posa) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala]: will work on it later

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/address/disposal_date.scala.html [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+@(form: Form[LocalDate], backLink: Call)(implicit request: RequestWithSessionData[_], messages:Messages, viewConfig: ViewConfig)
+
+@key = @{"multipleDisposalsDisposalDate"}
+@hasErrors = @{form.hasErrors}
+@title = @{messages(s"$key.title")}
+@errorKey = @{List("disposalDate-day", "disposalDate-month", "disposalDate-year", "disposalDate").map(form.error).find(_.isDefined).flatten}
+
+@mainTemplate(title = title, userType = request.userType, hasErrors = hasErrors, withSignOutLink = true) {
+
+ @backLinkComponent(backLink)
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/address/disposal_date.scala.html]: we can probably take this out for now while we don't know what the expected behaviour is here

File: conf/messages [hunk 5]
```diff
@@ -1275,6 +1278,45 @@ hasSupportingDocsToUpload.title=Do you want to upload any supporting docs?
 hasSupportingDocsToUpload.error.required=Select yes if you want to upload any supporting documents
 hasSupportingDocsToUpload.error.boolean=Select yes if you want to upload any supporting documents
 
+multipleDisposalsDisposalDate.title=What date did you exchange contracts when disposing of the property?
+multipleDisposalsDisposalDate.helpText=It’s usually when you and the buyer or new owner swapped contracts and the buyer paid a deposit. This is also called the ‘disposal’ date. For example, 25 4 2020
+multipleDisposalsDisposalDate.link=<a href="{0}" target="_blank">Find out what happens on the exchange date (opens in a new window)</a>
```
Comment (AndyHWChung) [inline conf/messages]: missing a full stop after the `</a>`

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**